includes/wikia/services/AvatarService.class.php

```diff
@@ -267,16 +267,18 @@ public static function getVignetteUrl( Masthead $masthead, $width, $timestamp )
 		return $url;
 	}
 
-	private static function vignetteCustomUrl( $width, $relativePath, $timestamp ) {
+	private static function vignetteCustomUrl( $width, $relativePath, $timestamp ) : string {
 		global $wgVignetteUrl;
 
-		$url = $relativePath;
 		if ( !preg_match( '/^https?:\/\//', $relativePath ) ) {
 			$bucket = VignetteRequest::parseBucket( "{$wgVignetteUrl}/common/avatars" );
 			$relativePath = ltrim( $relativePath, '/' );
 			$url = self::buildVignetteUrl( $width, $bucket, $relativePath, $timestamp );
 		}
 		else {
+			// SUS-4341 | something is setting "avatar" URLs with scaling parameters, remove them
+			$relativePath = preg_replace( '#(/revision/latest)?/scale-to-width-down/\d+$#', '', $relativePath );
```

Comment:
Did you consider using `VignetteUrlToUrlGenerator` https://github.com/Wikia/app/blob/730c673fc590952adba60f0b0d56f2e2c6cdd6a4/includes/wikia/vignette/VignetteUrlToUrlGenerator.php#L9-L9 instead of regexing? I'm not sure if we are able to handle this case, but I'd prefer to not introduce logic related to vignette URL parsing in the other places.

Reply:
It's a "hack" for broken avatar URLs. That's why I wanted to keep handling of this specific case here. PLATFORM or SERVICES will take care of migrating the data, so that we will have canonical avatar URLs only.